#  Actividad M2.1 Uso de Google Colab
### Emiliano Ferreira Guadarrama A1654418


Hola, Ferre :))

    /![Imagen de notebook de Sofi](https://drive.google.com/uc?id=1vNYpqY7dvgtGCrvx8DJC2LoYFFEvWn8_)

In [19]:
import pandas as pd

# Lectura de Datos

In [20]:
from google.colab import drive
import os
drive.mount('/content/drive')
os.chdir('/content/drive/MyDrive/S4/DS')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [21]:
df = pd.read_csv('titanic.csv')
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [22]:
df.tail()

PassengerId  Survived  Pclass                                      Name  \
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
886    male  27.0      0      0      211536  13.00   NaN        S  
887  female  19.0      0      0      112053  30.00   B42        S  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

# Descripción de Variables



1.   PassengerId: Numérica, entero único
2.   Survived: Categórica, booleana [0, 1]
3.   Pclass: Categórica, [1, 2, 3]
4.   Name: Categórica, unique
5.   Sex: Categórica, ['male', 'female']
6.   Age: Numérica
7.   SibSp: Categórica, [0, 1, 2, 3, 4, 5, 8]
8.   Parch: Categórica, [0, 1, 2, 3, 4, 5, 6]
9.   Ticket: Categórica
10.  Fare: Numérica
11.  Cabin: Categórica
12.  Embarked: Categórica ['S', 'C', 'Q', nan]



df.columns muestra los nombres de todas las columnas del dataframe

In [23]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

df.info() muestra el tipo de dato de cada columna, al igual que el número de entradas no-nulas. Por ejemplo podemos ver que la mayoría de las entradas no tienen un valor para la cabina.

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


df.describe() muestra el número de entradas no-nulas, el promedio, desviación estándar, mínimo, máximo y cuartiles de las variables numéricas.

In [25]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [26]:
na_count = sum(pd.isna(df).apply(any, axis=1))

print(
    f"En total hay {len(df)-1} registros.\nHay {na_count} entradas donde al menos un valor es NaN\nQuedan {len(df)-1-na_count} entradas donde ningún valor es NaN."
)

En total hay 890 registros.
Hay 708 entradas donde al menos un valor es NaN
Quedan 182 entradas donde ningún valor es NaN.


Las columnas que aparecen en df.describe() son las numéricas, como fue mencionado anteriormente. Podemos ver que:


*   La tarifa más cara es 512.3292
*   El promedio de edad es de 29.69911764705882





In [27]:
df.describe().loc[["max"], ["Fare"]]

Fare
max  512.3292

In [28]:
df.describe().loc[["mean"], ["Age"]]

Age
mean  29.699118

In [29]:
categoricals = df.iloc[:, [1, 2, 3, 4, 6, 7, 8, 10, 11]]

for colname, count in [(colname, len(column.unique())) for colname, column in categoricals.iteritems()]:
  print(f"Column {colname:^8} has {count:>3}\tunique values")

Column Survived has   2	unique values
Column  Pclass  has   3	unique values
Column   Name   has 891	unique values
Column   Sex    has   2	unique values
Column  SibSp   has   7	unique values
Column  Parch   has   7	unique values
Column  Ticket  has 681	unique values
Column  Cabin   has 148	unique values
Column Embarked has   4	unique values


# Análisis de Datos

In [30]:
f"En total se pagaron ${sum(df.Fare):,.2f} en tarifas de los pasajeros"

'En total se pagaron $28,693.95 en tarifas de los pasajeros'

In [31]:
sobrevivieron    = df[df.Survived == 1]
no_sobrevivieron = df[df.Survived == 0]
sobrevivieron.head()

PassengerId  Survived  Pclass  \
1            2         1       1   
2            3         1       3   
3            4         1       1   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

In [32]:
no_sobrevivieron.head()

PassengerId  Survived  Pclass                            Name   Sex   Age  \
0            1         0       3         Braund, Mr. Owen Harris  male  22.0   
4            5         0       3        Allen, Mr. William Henry  male  35.0   
5            6         0       3                Moran, Mr. James  male   NaN   
6            7         0       1         McCarthy, Mr. Timothy J  male  54.0   
7            8         0       3  Palsson, Master. Gosta Leonard  male   2.0   

   SibSp  Parch     Ticket     Fare Cabin Embarked  
0      1      0  A/5 21171   7.2500   NaN        S  
4      0      0     373450   8.0500   NaN        S  
5      0      0     330877   8.4583   NaN        Q  
6      0      0      17463  51.8625   E46        S  
7      3      1     349909  21.0750   NaN        S

In [34]:
!apt install texlive-xetex texlive-fonts-recommended texlive-generic-recommended
!jupyter nbconvert M2-1-EmilianoFerreira.ipynb --to pdf

Reading package lists... Done
Building dependency tree       
Reading state information... Done
texlive-fonts-recommended is already the newest version (2017.20180305-1).
texlive-generic-recommended is already the newest version (2017.20180305-1).
texlive-xetex is already the newest version (2017.20180305-1).
The following package was automatically installed and is no longer required:
  libnvidia-common-460
Use 'apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 42 not upgraded.
[NbConvertApp] Converting notebook M2-1-EmilianoFerreira.ipynb to pdf
[NbConvertApp] Writing 66710 bytes to ./notebook.tex
[NbConvertApp] Building PDF
[NbConvertApp] Running xelatex 3 times: ['xelatex', './notebook.tex', '-quiet']
[NbConvertApp] Running bibtex 1 time: ['bibtex', './notebook']
[NbConvertApp] WARNING | bibtex had problems, most likely because there were no citations
[NbConvertApp] PDF successfully created
[NbConvertApp] Writing 61569 bytes to M2-1-EmilianoFerreira.pdf
